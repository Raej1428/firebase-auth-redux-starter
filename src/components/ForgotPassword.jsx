// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import { withFormik } from 'formik';
import Yup from 'yup';
import TextInput from './elements/TextInput';
import FormButton from './elements/FormButton';
import FormWideError from './elements/FormWideError';
import FormWideNotification from './elements/FormWideNotification';

/*
const PasswordForgetPage = ({ sendPasswordResetEmail }: { sendPasswordResetEmail: Function }) =>
  <PasswordForgetForm sendPasswordResetEmail={sendPasswordResetEmail} />;
*/

// Our inner form component which receives our form's state and updater methods as props
const InnerForm = ({
  values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting, status,
}: {
    values: Object,
    errors: Object,
    touched: Object,
    handleChange: Function,
    handleBlur: Function,
    handleSubmit: Function,
    isSubmitting: boolean,
    status: string,
  }) =>
  (
    <main className="pa4 black-80">
      <form onSubmit={handleSubmit} className="measure center">
        <fieldset id="sign_up" className="ba b--transparent ph0 mh0">
          <legend className="f4 fw6 ph0 mh0">Reset my password</legend>
          <FormWideError error={errors.formwide} />
          <FormWideNotification message={status} />
          <div className="mt3">
            <TextInput
              id="email"
              type="text"
              label="Email"
              error={touched.email && errors.email}
              value={values.email}
              onChange={handleChange}
              onBlur={handleBlur}
            />
          </div>
        </fieldset>
        <FormButton
          disabled={Object.keys(errors).length !== 0 || isSubmitting}
          text="Reset Password"
        />
        <div className="lh-copy mt3">
          <Link to="/signup">
            <div className="f6 link dim black db">Don&apos;t have an account?</div>
          </Link>
          <Link to="/signin">
            <div className="f6 link dim black db">Sign In</div>
          </Link>
        </div>
      </form >
    </main >
  );

// Wrap our form with the using withFormik HoC
const PasswordForgetForm = withFormik({
  // Transform outer props into form values
  mapPropsToValues: () => ({ email: '' }),
  // Add a custom validation function (this can be async too!)
  validationSchema: Yup.object().shape({
    email: Yup.string()
      .email('Invalid email address')
      .required('Email is required!'),
  }),
  // Submission handler
  handleSubmit: (values, { props, setSubmitting }) => {
    props.sendPasswordResetEmail(values.email)
      .then(() => {
        setSubmitting(false);
        // setStatus('Email sent. Check your inbox');
        props.sendSuccessMessage('Email sent. Check your inbox');
      })
      .catch(() => {
        // Dont tell them if account exists or not.
        setSubmitting(false);
        // setStatus('Email sent. Check your inbox');
        props.sendSuccessMessage('Email sent. Check your inbox');
      });
  },
})(InnerForm);


export default PasswordForgetForm;
