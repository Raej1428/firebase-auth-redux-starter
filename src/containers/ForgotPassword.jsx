// @flow
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { sendPasswordResetEmail } from '../actions/auth';
import ForgotPassword from '../components/ForgotPassword';
import { addSuccess } from '../actions/toast';

const mapStateToProps = state => ({
  isAuthenticated: state.auth.authenticated,
});

const mapDispatchToProps = dispatch => ({
  sendPasswordResetEmail: email => sendPasswordResetEmail(email),
  sendSuccessMessage: message => dispatch(addSuccess(message)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(ForgotPassword));
