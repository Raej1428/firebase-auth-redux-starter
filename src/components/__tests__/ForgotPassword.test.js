/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import ForgotPassword from '../ForgotPassword';
import { BrowserRouter as Router } from 'react-router-dom';

describe('ForgotPassword component', () => {
  it('matches snapshot ', () => {
    const wrapper = shallow(
      <ForgotPassword />
    );
    expect(wrapper).toMatchSnapshot();
  })
})
