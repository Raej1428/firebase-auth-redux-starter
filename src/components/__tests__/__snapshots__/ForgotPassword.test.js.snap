// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component matches snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <C />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": false,
      "handleSubmit": [Function],
      "initialValues": Object {
        "email": "",
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "render": [Function],
      "validate": undefined,
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": false,
        "handleSubmit": [Function],
        "initialValues": Object {
          "email": "",
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "render": [Function],
        "validate": undefined,
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
